File: 0070-climbing-stairs/0070-climbing-stairs.py
from functools import cache;
class Solution:
    @cache
    def climbStairs(self, n: int) -> int:
        waysToClimb = [0,1,2]
        if n==1:
            return 1
        if n==2:
            return 2
        if n>2:
            return self.climbStairs(n-1) + self.climbStairs(n-2)

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    const waysToClimb = [0,1,2];
    for (let i = 3; i <= n; i++) {
        waysToClimb.push(waysToClimb[i-1] + waysToClimb[i-2])
    }
    return waysToClimb[n]
};